pkg/index/index_factory_test.go

```diff
@@ -0,0 +1,199 @@
+package index_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/buildpacks/imgutil"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/random"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/heroku/color"
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+
+	"github.com/buildpacks/pack/pkg/index"
+	h "github.com/buildpacks/pack/testhelpers"
+)
+
+var dockerRegistry *h.TestRegistryConfig
+
+func TestIndexFactory(t *testing.T) {
+	color.Disable(true)
+	defer color.Disable(false)
+
+	h.RequireDocker(t)
+
+	dockerRegistry = h.RunRegistry(t)
+	defer dockerRegistry.StopRegistry(t)
+
+	os.Setenv("DOCKER_CONFIG", dockerRegistry.DockerConfigDir)
+	spec.Run(t, "Fetcher", testIndexFactory, spec.Parallel(), spec.Report(report.Terminal{}))
+}
+
+func testIndexFactory(t *testing.T, when spec.G, it spec.S) {
+	var (
+		indexFactory  *index.IndexFactory
+		imageIndex    imgutil.ImageIndex
+		indexRepoName string
+		err           error
+		tmpDir        string
+	)
+
+	it.Before(func() {
+		tmpDir, err = os.MkdirTemp("", "index-factory-test")
+		h.AssertNil(t, err)
+		indexFactory = index.NewIndexFactory(authn.DefaultKeychain, tmpDir)
+	})
+
+	it.After(func() {
+		os.RemoveAll(tmpDir)
+	})
+
+	when("#CreateIndex", func() {
+		it.Before(func() {
+			indexRepoName = h.NewRandomIndexRepoName()
+		})
+
+		when("no options are provided", func() {
+			it("creates an image index", func() {
+				imageIndex, err = indexFactory.CreateIndex(indexRepoName)
+				h.AssertNil(t, err)
+				h.AssertNotNil(t, imageIndex)
+			})
+		})
+	})
+
+	when("#Exists", func() {
+		when("index exists on disk", func() {
+			it.Before(func() {
+				indexRepoName = h.NewRandomIndexRepoName()
+				setUpLocalIndex(t, indexFactory, indexRepoName)
+			})
+
+			it("returns true", func() {
+				h.AssertTrue(t, indexFactory.Exists(indexRepoName))
+			})
+		})
+
+		when("index does not exist on disk", func() {
+			it.Before(func() {
+				indexRepoName = h.NewRandomIndexRepoName()
+			})
+
+			it("returns false", func() {
+				h.AssertFalse(t, indexFactory.Exists(indexRepoName))
+			})
+		})
+	})
+
+	when("#LoadIndex", func() {
+		when("index exists on disk", func() {
+			it.Before(func() {
+				indexRepoName = h.NewRandomIndexRepoName()
+				setUpLocalIndex(t, indexFactory, indexRepoName)
+			})
+
+			it("loads the index from disk", func() {
+				imageIndex, err = indexFactory.LoadIndex(indexRepoName)
+				h.AssertNil(t, err)
+				h.AssertNotNil(t, imageIndex)
```

Comment:
Similar comment to https://github.com/buildpacks/pack/pull/1705/files#r1591429858